tasks/managed/extract-index-image/extract-index-image.yaml

```diff
@@ -112,11 +112,11 @@ spec:
 
         RESULTS_FILE="$(params.dataDir)/$(params.resultsDirPath)/extract-index-image-results.json"
 
-        indexImage="$(jq -r '.components[-1].index_image' \
+        indexImage="$(jq -r '.components[].index_image' \
                     "$(params.dataDir)/$(params.internalRequestResultsFile)")"
         echo -n "$indexImage" | tee "$(results.indexImage.path)"
 
-        indexImageResolved="$(jq -r '.components[-1].index_image_resolved' \
+        indexImageResolved="$(jq -r '.components[].index_image_resolved' \
```

Comment:
When you have multiple fragments shipping to a single OCP version, I think that this results in improper logic. When there are multiple fragments in a single OCP version, then the _last_ resolved index would be the correct one because of how they are chained together.

The original bug wants to show all OCP versions, but that is when there is only one fragment per OCP version. 

I think we might need to group by OCP version, preserving the order, and then get the last one for each.

Reply:
Do we want to change the "index_image" field to be keyed by OCP version? Something like this?
```json
{
    "index_image": {
      "v4.14": {
        "index_image": "registry-proxy.engineering.redhat.com/rh-osbs/iib:1014428",
        "index_image_resolved": "registry-proxy.engineering.redhat.com/rh-osbs/iib@sha256:abc123..."
      },
      "v4.15": {
        "index_image": "registry-proxy.engineering.redhat.com/rh-osbs/iib:1014429",
        "index_image_resolved": "registry-proxy.engineering.redhat.com/rh-osbs/iib@sha256:def456..."
      }
    }
  }
```

Reply:
Yes, that looks like it would be beneficial.